service/core.go

```diff
@@ -193,3 +199,21 @@ func (c *Core) nextTaskID() int64 {
 func (c *Core) requestLogger(r *http.Request) *zap.Logger {
 	return c.logger.With(zap.String("request_id", api.RequestIDFromContext(r.Context())))
 }
+
+func (c *Core) publishEvent(event string, data string) {
+	eventPayload := fmt.Sprintf("event: %s\n", event)
+	dataLines := strings.Split(data, "\n")
+
+	for _, line := range dataLines {
+		eventPayload += fmt.Sprintf("data: %s\n", line)
+	}
+
+	go func() {
+		c.subscriptionsMu.RLock()
+		defer c.subscriptionsMu.RUnlock()
+
+		for sub := range c.subscriptions {
+			sub <- []byte(eventPayload + "\n")
+		}
```

Comment:
No need for the defer.
```suggestion
		c.subscriptionsMu.RLock()
		for sub := range c.subscriptions {
			sub <- []byte(eventPayload + "\n")
		}
		c.subscriptionsMu.RUnLock()
```